pkg/controller/cloudfront/distribution/setup.go

```diff
@@ -323,9 +323,13 @@ func postUpdate(_ context.Context, cr *svcapitypes.Distribution, resp *svcsdk.Up
 func preUpdate(_ context.Context, cr *svcapitypes.Distribution, udi *svcsdk.UpdateDistributionInput) error {
 	udi.Id = awsclients.String(meta.GetExternalName(cr))
 	udi.SetIfMatch(awsclients.StringValue(cr.Status.AtProvider.ETag))
-	udi.DistributionConfig.CallerReference = awsclients.String(string(cr.UID))
+	if *cr.Spec.ForProvider.CallerReference != "" { // import scenario
```

Comment:
One point to note is that because we are adding this logic to only `preUpdate`, we are not supporting the use case of explicitly specifying a caller reference by the user of the API, i.e., for newly provisioned CloudFront Distributions, caller ref. will always be the `uid`, although we have exposed an (internal) API for it. I think this is not an important use case (as long as, for instance, AWS decides to impose a (backward-incompatible) format restriction on the caller ref. string). But I think, we should at least provide field documentation mentioning this. Is it possible to introduce this field only for the `status` kind?

Reply:
I updated `preCreate` function to use specified `callerReference` if exists. Also, extended the documentation.

Reply:
Thanks @ezgidemirel!